### PR TITLE
Wait for the Lambda functions to stabilize in the create/update workf…

### PR DESCRIPTION
**Pull request by abrooksv**
…lows

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/SdkWaiterUtils.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+package software.aws.toolkits.jetbrains.utils
+
+import software.amazon.awssdk.core.waiters.WaiterResponse
+
+fun <T> WaiterResponse<T>.response(): T {
+    val responseOrException = this.matched()
+    if (responseOrException.response().isPresent) {
+        return responseOrException.response().get()
+    }
+    if (responseOrException.exception().isPresent) {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/utils/SdkWaiterUtils.kt:13]: Should we check exception first? (is it possible to have both?)

File: core/tst/software/aws/toolkits/core/utils/DelegateSdkConsumers.kt [hunk 2]
```diff
@@ -19,27 +20,34 @@ import kotlin.reflect.full.isSubclassOf
  *
  * This will handle "simple" methods that they generate as well such as ListBuckets that takes 0 arguments.
  */
-class DelegateSdkConsumers : Answer<Any> {
+class DelegateSdkConsumers(private val sdkClass: Class<*>) : Answer<Any> {
     override fun answer(invocation: InvocationOnMock): Any? {
         val method = invocation.method
-        return if (method.isDefault &&
-            method?.parameters?.getOrNull(0)?.type?.kotlin?.isSubclassOf(SdkRequest::class) != true &&
-            method.name !in setOf("utilities", "waiter") // Never call the real methods they use to get to their helper classes since it always throws
```
Comment (hunterwerlla) [inline core/tst/software/aws/toolkits/core/utils/DelegateSdkConsumers.kt:27]: `utilities` was deleted from this logic, is that OK?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/upload/steps/UpdateLambdaCodeTest.kt [hunk 3]
```diff
@@ -73,6 +76,9 @@ class UpdateLambdaCodeTest {
         val lambdaClient = clientManagerRule.create<LambdaClient>().stub {
             on { updateFunctionCode(codeRequestCaptor.capture()) } doReturn UpdateFunctionCodeResponse.builder().build()
             on { updateFunctionConfiguration(configRequestCaptor.capture()) } doReturn UpdateFunctionConfigurationResponse.builder().build()
+            on { getFunctionConfiguration(any<GetFunctionConfigurationRequest>()) } doReturn GetFunctionConfigurationResponse.builder()
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/upload/steps/UpdateLambdaCodeTest.kt:79]: Nope, only create does: https://github.com/aws/aws-sdk-java-v2/blob/master/services/lambda/src/main/resources/codegen-resources/waiters-2.json#L12

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/SdkWaiterUtils.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/utils/SdkWaiterUtils.kt:13]: It is not according to matt

File: core/tst/software/aws/toolkits/core/utils/DelegateSdkConsumers.kt (hunk 2 above)
Comment (abrooksv) [reply core/tst/software/aws/toolkits/core/utils/DelegateSdkConsumers.kt:27]: `software.aws.toolkits.jetbrains.services.s3.S3VirtualBucketTest#getUrl` still passes and it was added for that test so I don't think so

Comment (hunterwerlla) [reply core/tst/software/aws/toolkits/core/utils/DelegateSdkConsumers.kt:27]: Cool

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**